WeakAuras/Prototypes.lua

```diff
@@ -4393,159 +4393,246 @@ WeakAuras.event_prototypes = {
     events = function(trigger)
       local result = {
         "UNIT_SPELLCAST_CHANNEL_START",
-        "UNIT_SPELLCAST_CHANNEL_STOP",
-        "UNIT_SPELLCAST_CHANNEL_UPDATE",
         "UNIT_SPELLCAST_START",
-        "UNIT_SPELLCAST_STOP",
         "UNIT_SPELLCAST_DELAYED",
+        "UNIT_SPELLCAST_CHANNEL_UPDATE",
         "UNIT_SPELLCAST_INTERRUPTIBLE",
         "UNIT_SPELLCAST_NOT_INTERRUPTIBLE",
+        "UNIT_SPELLCAST_STOP",
+        "UNIT_SPELLCAST_CHANNEL_STOP",
+        "UNIT_SPELLCAST_INTERRUPTED",
+        "UNIT_SPELLCAST_FAILED",
+        "UNIT_SPELLCAST_FAILED_QUIET",
       };
-      AddUnitChangeEvents(trigger.unit, result);
-      return result;
+      AddUnitChangeEvents(trigger.unit, result)
+      if trigger.target ~= "" then
```

Comment:
I don't undertand this code. What is trigger.target?

Reply:
trigger.target reference a field i added that let you specify who is the target of the cast.
This give the possibility to use the generic cast trigger for example to show any cast with you as the target (with unit=multi and target=player) or when Zek'voz eye target's you (unit=boss2 and target=player)
There are quite a few npc spells that doesn't target the right destination when they start casting, and change target during the cast, so in this part of the code i added the event "UNIT_TARGET" for checking the trigger (in uldir the zkvoz' eye beam and zul's shadow barrage for ex)

Reply:
Ah, hmm, but wouldn't they untrigger as soon as the boss selects a different target?

I'm a bit unsure if the feature will work good enough to not require tons of explanation on why it doesn't work "correctly"

Reply:
> wouldn't they untrigger as soon as the boss selects a different target?

Yes it will untrigger if boss change target, or trigger after the cast has started if trigger.target is not the first target during the cast, but that's enough to run a sound alert on show or whatever, and it will always show an accurate timer for the end of the cast.

For eyebeam, bigwigs register event "UNIT_TARGET" at the start of the cast and run it's "printTarget()" on next UNIT_TARGET event with the right source. It works for this case but is not generic enough.
For DBM after SPELL_CAST_START it check target every 0.1s